crates/sparrow-arrow/src/downcast.rs

```diff
@@ -56,6 +49,14 @@ pub fn downcast_struct_array(array: &dyn Array) -> anyhow::Result<&StructArray>
         .with_context(|| format!("Unable to downcast {:?} to struct array", array.data_type()))
 }
 
+/// Downcast an `ArrayRef` to a `ListArray`.
+pub fn downcast_list_array(array: &dyn Array) -> anyhow::Result<&ListArray> {
```

Comment:
Thought: We probably don't need this method (based on our other discussions around just using the Arrow cast methods). Maybe see if we can get by without it?

Reply:
Yeah, can do so. Was just going to keep the same pattern until we update them all at once, but might as well do it now and less work later.

Reply:
It looks like the method is no longer used -- want to delete it?

Reply:
It was used in a few places but moved those to `as_list` as well. Deleted